pkg/validate/csv.go

```
@@ -163,3 +167,94 @@ func isEmptyValue(v reflect.Value) bool {
 		panic(fmt.Sprintf("%v kind is not supported.", v.Kind()))
 	}
 }
+
+// validateExamplesAnnotations compares alm/olm example annotations with provided APIs given
+// by Spec.CustomResourceDefinitions.Owned and Spec.APIServiceDefinitions.Owned.
+func validateExamplesAnnotations(csv v1alpha1.ClusterServiceVersion) (manifestResult validator.ManifestResult) {
+	var examples []v1beta1.CustomResourceDefinition
+	var annotationsNames = []string{"alm-examples", "olm.examples"}
+	var annotationsExamples string
+	var ok bool
+	annotations := csv.ObjectMeta.GetAnnotations()
+	// Return right away if no examples annotations are found.
+	if annotations == nil {
+		manifestResult.Warnings = append(manifestResult.Warnings, validator.InvalidOperation(fmt.Sprintf("Warning: example annotations not found for %s", csv.GetName()), csv.GetName()))
+		return
+	}
+	// Expect either `alm-examples` or `old.examples` but not both
```

If you cannot use both simultaneously, there should be an error returned if they are.

---

I'm not sure if that should raise an error. I can return a warning instead if mentioning both would not affect the operator.

---

Either return a warning or change the comment.

---

Sure.